### DISCUSSION TITLE
Instanciar cronômetros diferentes na mesma página (javascript)

### DISCUSSION THREAD

**Discussion by reynaldobarrosjr** [frontendbr/forum#1075] [Perguntas e Respostas]
Cara
, usando o Easytimer de Albert Gonzales - https://albert-gonzalez.github.io/easytimer.js/
fiz um teste para instanciar vários cronômetros com inicios de contagem aleatórios.

No código que vou compartilhar faço isso de forma tosca...a  partir da linha 23. 
https://gist.github.com/trevobr/37b7e9947a33a3a939ff154f2e7e178a

Queria saber a forma 'elegante' de fazer o mesmo.  

Você que é ninja no javascript, pode me ajudar?

p.s: Coloquei o código zipado aqui também. 

[cronometros.zip](https://github.com/frontendbr/forum/files/1989466/cronometros.zip)

Desde já agradeço a ajuda.

**Comment by willycamargo** [comment]
Não entendi muito bem sua necessidade, mas acho que eu faria algo nesse sentido:

https://jsfiddle.net/e4pe3q9t/

```html
<div id="callbackExample0">
    <span>Timer 0</span>
    <h1>
        <div class="values"></div>
    </h1>
</div>
<div id="callbackExample1">
    <span>Timer 1</span>
    <h1>
        <div class="values"></div>
    </h1>
</div>
<div id="callbackExample2">
    <span>Timer 2</span>
    <h1>
        <div class="values"></div>
    </h1>
</div>
<div id="callbackExample3">
    <span>Timer 3</span>
    <h1>
        <div class="values"></div>
    </h1>
</div>
<div id="callbackExample4">
    <span>Timer 4</span>
    <h1>
        <div class="values"></div>
    </h1>
</div>
```

```javascript
function makeTimer(selector, seconds = Math.floor(Math.random() * 20)) {
    // setup
    var $element = document.querySelector(selector);
    var timer = new Timer();
    timer.start({
        startValues: {
            seconds: seconds
        }
    });

    $element.innerHTML = timer.getTimeValues().toString();
    var interval = setInterval(function() {
        $element.innerHTML = timer.getTimeValues().toString();
    }, 1000);
}

makeTimer('#callbackExample0 .values', 0);
makeTimer('#callbackExample1 .values', 10);
makeTimer('#callbackExample2 .values', 20);
makeTimer('#callbackExample3 .values', 30);
makeTimer('#callbackExample4 .values', 40);
```

Ainda seria possível fazer um retorno para poder controlar pause, reset, stop etc...
Mas ai tem que ver a necessidade.

obs: removi o JQuery das dependências e estou considerando que o script será inserido no footer do seu `<body>`

-----------

**UPDATE:** Dei uma lida melhor na documentação e melhorei o JS para ficar mais simples (sem `setInterval`) e atender possíveis necessidades de controles:

https://jsfiddle.net/camargowilly/e4pe3q9t/3/

**Comment by reynaldoapb** [comment]
Mano ...fiquei feliz com a resposta...vou olhar quando chegar em casa ;) e te dou um retorno. vocês são feras!
